service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/ProcedureRequestMapper.java

```diff
@@ -0,0 +1,36 @@
+package uk.nhs.adaptors.oneoneone.cda.report.mapper;
+
+import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
+
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Coding;
+import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.stereotype.Component;
+
+import lombok.RequiredArgsConstructor;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01ClinicalDocument1;
+
+@Component
+@RequiredArgsConstructor
+public class ProcedureRequestMapper {
+    public ProcedureRequest mapProcedureRequest(POCDMT000002UK01ClinicalDocument1 clinicalDocument) {
+        ProcedureRequest procedureRequest = new ProcedureRequest();
+        if (clinicalDocument.isSetComponentOf()) {
+            if (clinicalDocument.getComponentOf().getEncompassingEncounter() != null) {
+                if (clinicalDocument.getComponentOf().getEncompassingEncounter().isSetDischargeDispositionCode()) {
+                    if (clinicalDocument.getComponentOf().getEncompassingEncounter().getDischargeDispositionCode().isSetDisplayName()) {
+                        procedureRequest.setIdElement(newRandomUuid());
+                        String procedureReason =
+                            clinicalDocument.getComponentOf().getEncompassingEncounter().getDischargeDispositionCode().getDisplayName();
```

Comment:
Please see the following comment:
"Disposition is located at /ClinicalDocument/componentOf/encompassingEncounter/discharge DispositionCode/@code – the code system is fixed to “2.16.840.1.113883.2.1.3.2. 4.17.325”
This should be mapped to ReferralRequest.supportingInfo(ProcedureRequest).code"

Let's map full CodeableConcept -  system, code, display